internal/database/repository/application_int_test.go

```diff
@@ -0,0 +1,268 @@
+		{
+			name:     "Filter By Group and User",
+			filters:  ApplicationFilters{Groups: []string{"group1", "group2"}, User: util.ToPtr("user1")},
+			expected: 2,
+		},
+		{
+			name:     "No Filters",
+			expected: 6,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			apps, err := repo.GetAllApplications(context.Background(), tt.filters)
+			require.NoError(t, err)
+			assert.Equal(t, tt.expected, len(apps))
+		})
+	}
+}
+
+func TestGetAppsPerPartitionPerQueue_Integration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test in short mode")
+	}
+
+	ctx := context.Background()
+
+	connPool := database.NewTestConnectionPool(ctx, t)
+
+	repo, err := NewPostgresRepository(connPool)
+	if err != nil {
+		t.Fatalf("could not create repository: %v", err)
+	}
+
+	seedApplications(t, repo)
+
+	tests := []struct {
+		name     string
+		filters  ApplicationFilters
+		expected int
+	}{
+		{
+			name: "Filter by User",
+			filters: ApplicationFilters{
+				User: util.ToPtr("user1"),
+			},
+			expected: 3,
+		},
+		{
+			name: "Filter by Submission Time Range",
+			filters: ApplicationFilters{
+				SubmissionStartTime: util.ToPtr(time.Now().Add(-3 * time.Hour)),
+				SubmissionEndTime:   util.ToPtr(time.Now().Add(-1 * time.Hour)),
+			},
+			expected: 2,
+		},
+		{
+			name: "Filter by Finished Time Range",
+			filters: ApplicationFilters{
+				FinishedStartTime: util.ToPtr(time.Now().Add(-2 * time.Hour)),
+				FinishedEndTime:   util.ToPtr(time.Now().Add(-1 * time.Hour)),
+			},
+			expected: 1,
+		},
+		{
+			name: "Filter By Limit",
+			filters: ApplicationFilters{
+				Limit: util.ToPtr(2),
+			},
+			expected: 2,
+		},
+		{
+			name: "Filter By Limit and Offset",
+			filters: ApplicationFilters{
+				Limit:  util.ToPtr(2),
+				Offset: util.ToPtr(5),
+			},
+			expected: 1,
+		},
+		{
+			name:     "Filter By Group",
+			filters:  ApplicationFilters{Groups: []string{"group1", "group2"}},
+			expected: 4,
+		},
+		{
+			name:     "Filter By Group and User",
+			filters:  ApplicationFilters{Groups: []string{"group1", "group2"}, User: util.ToPtr("user1")},
+			expected: 2,
+		},
+		{
+			name:     "No Filters",
+			expected: 6,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			apps, err := repo.GetAppsPerPartitionPerQueue(context.Background(), "default", "root.default", tt.filters)
+			require.NoError(t, err)
+			assert.Equal(t, tt.expected, len(apps))
+		})
+	}
+}
+
+func seedApplications(t *testing.T, repo *PostgresRepository) {
+	t.Helper()
+
+	now := time.Now()
+
+	apps := []*dao.ApplicationDAOInfo{
+		{
+			ApplicationID:   "app1",
+			UsedResource:    map[string]int64{"cpu": 1},
+			MaxUsedResource: map[string]int64{"cpu": 2},
+			PendingResource: map[string]int64{"cpu": 1},
+			Partition:       "default",
+			QueueName:       "root.default",
+			SubmissionTime:  now.Add(-6 * time.Hour).UnixMilli(),
+			User:            "user1",
+			State:           "RUNNING",
```

Comment:
What do you think about putting all the states in a set of string constants? Or maybe yunikorn already exports them?

Reply:
I just looked at the `yunikorn-scheduler-interface` proto definitions - they define state changes like `EventRecord_APP_RUNNING` - see `lib/go/si/si.pb.go`.  Think it's worth it to use those?

Reply:
addressed